modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/enricher/ExtensionsErrorsDeclarationEnricher.java

```diff
@@ -36,43 +42,63 @@
  *
  * @since 4.0
  */
-public class ConnectionErrorsDeclarationEnricher implements DeclarationEnricher {
+public class ExtensionsErrorsDeclarationEnricher implements DeclarationEnricher {
 
-  private static final String MULE_NAMESPACE = CORE_PREFIX.toUpperCase();
+  public static final String MULE = CORE_NAMESPACE_NAME;
 
   @Override
   public void enrich(ExtensionLoadingContext extensionLoadingContext) {
-    ExtensionDeclaration declaration = extensionLoadingContext.getExtensionDeclarer().getDeclaration();
-    Optional<ImplementingTypeModelProperty> implementingType = declaration.getModelProperty(ImplementingTypeModelProperty.class);
+    ExtensionDeclaration extensionDeclaration = extensionLoadingContext.getExtensionDeclarer().getDeclaration();
+    Optional<ImplementingTypeModelProperty> implementingType =
+        extensionDeclaration.getModelProperty(ImplementingTypeModelProperty.class);
 
     if (implementingType.isPresent()) {
       ExtensionElement extensionElement = new ExtensionTypeWrapper<>(implementingType.get().getType());
       extensionElement.getAnnotation(ErrorTypes.class).ifPresent(errorTypesAnnotation -> {
 
-        Set<ErrorModel> errorModels = declaration.getErrorModels();
+        Set<ErrorModel> errorModels = extensionDeclaration.getErrorModels();
         if (!errorModels.isEmpty()) {
           new IdempotentDeclarationWalker() {
 
             @Override
-            public void onOperation(WithOperationsDeclaration owner, OperationDeclaration operation) {
-              if (operation.getModelProperty(ConnectivityModelProperty.class).isPresent()) {
-                operation.addError(getErrorModel(CONNECTIVITY, errorModels, operation));
-                operation.addError(getErrorModel(RETRY_EXHAUSTED, errorModels, operation));
+            public void onOperation(WithOperationsDeclaration owner, OperationDeclaration operationDeclaration) {
+              if (operationDeclaration.getModelProperty(ConnectivityModelProperty.class).isPresent()) {
+                operationDeclaration.addError(getErrorModel(CONNECTIVITY, errorModels, operationDeclaration));
+                operationDeclaration.addError(getErrorModel(RETRY_EXHAUSTED, errorModels, operationDeclaration));
               }
             }
-          }.walk(declaration);
+
+            @Override
+            protected void onSource(WithSourcesDeclaration owner, SourceDeclaration sourceDeclaration) {
+              if (sourceDeclaration.getSuccessCallback().isPresent() || sourceDeclaration.getErrorCallback().isPresent()) {
+                ErrorModel anyError = getMuleErrorModel(ANY, errorModels)
+                    .orElseThrow(() -> new IllegalStateException("Unable to retrieve the MULE:ANY error"));
+
+                ErrorModel sourceError = newError(SOURCE.getType(), MULE).withParent(anyError).build();
```

Comment:
Why is it like this? I was expecting to see two errors being added: one saying that the response couldn't be generated, and another saying that the response couldn't be sent (the last one, only if the source emits response). 

A unique and generic source error type isn't of much help to the user.